### PR TITLE
refactor/episode_details

### PR DESCRIPTION
**Pull request by banankmh**

### CODE REVIEW

File: feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/episode_details/BindingAdapters.kt [hunk 1]
```diff
@@ -24,16 +24,17 @@ fun <T> View.hideWhenNoProductionCode(productionCode:String){
...
+fun <T> View.hideWhenNoResultText(text:String){
+    if (text.isEmpty()){
+        this.visibility = View.GONE
+    }else{
+        this.visibility = View.VISIBLE
     }
 }
 
-@BindingAdapter("app:refreshing")
+@BindingAdapter("app:setRefreshing")
```
Comment (MohamedElgohary88) [inline feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/episode_details/BindingAdapters.kt:37]: delete useless functions

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/episode_details/EpisodeDetailsListener.kt [hunk 2]
```diff
@@ -5,5 +5,8 @@ import com.chocolatecake.bases.BaseInteractionListener
 interface EpisodeDetailsListener : BaseInteractionListener {
     fun clickToBack()
     fun clickToRate(episodeId: Int)
+    fun clickToPlayFullScreen(videoKey: String)
+
```
Comment (MohamedElgohary88) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/episode_details/EpisodeDetailsListener.kt]: white space :)

**Review by MohamedElgohary88 — APPROVED** [review]

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/episode_details/EpisodeDetailsListener.kt (hunk 2 above)
Comment (amna97alaa) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/episode_details/EpisodeDetailsListener.kt]: white spaces

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/episode_details/EpisodeDetailsUiEvent.kt [hunk 3]
```diff
@@ -5,5 +5,6 @@ sealed interface EpisodeDetailsUiEvent {
     data class ClickToRate(val episodeId: Int) : EpisodeDetailsUiEvent
     data class ClickCast(val itemId: Int) : EpisodeDetailsUiEvent
     data class SubmitRating(val message: String) : EpisodeDetailsUiEvent
+    data class ClickToPlayFullScreen(val videoKey:String) : EpisodeDetailsUiEvent
```
Comment (amna97alaa) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/episode_details/EpisodeDetailsUiEvent.kt]: white space

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/episode_details/EpisodeDetailsViewModel.kt [hunk 4]
```diff
@@ -68,11 +71,33 @@ class EpisodeDetailsViewModel @Inject constructor(
...
+    }
+
+    private fun onSuccessEpisodeVideo(trailerUiState: TrailerUiState) {
+        _state.update {
+            it.copy(
+                trailerKey = trailerUiState.videoKey,
+                refreshing = false, onErrors = emptyList()
             )
         }
+
```
Comment (amna97alaa) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/episode_details/EpisodeDetailsViewModel.kt]: white space

**Review by amna97alaa — CHANGES_REQUESTED** [review]
delete white spaces

**Pull request closed without merging**